### PR TITLE
Speed up some tests

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

**Comment by nouiz** [comment]
trigger with new master.

File: theano/tensor/extra_ops.py [hunk 1]
```diff
@@ -311,57 +320,51 @@ def c_code(self, node, name, inames, onames, sub):
...
 
-        if self.axis is None or (self.axis == 0 and node.inputs[0].ndim == 1):
-            code = """
+        code = """
+                int axis = %(params)s->c_axis;
+                if (axis == 0 && PyArray_NDIM(%(x)s) == 1)
+                    axis = NPY_MAXDIMS;
                 npy_intp shape[1] = { PyArray_SIZE(%(x)s) };
-                if(!(%(z)s && PyArray_DIMS(%(z)s)[0] == shape[0]))
+                if(axis == NPY_MAXDIMS && !(%(z)s && PyArray_DIMS(%(z)s)[0] == shape[0]))
```
Comment (notoraptor) [inline theano/tensor/extra_ops.py]: I think this does not check that %(z)s is a 1-D tensor (`PyArray_NDIM(%(z)s) == 1`), it only checks that z[0] has the expected shape.

File: theano/tensor/extra_ops.py [hunk 2]
```diff
@@ -311,57 +320,51 @@ def c_code(self, node, name, inames, onames, sub):
...
-                       %(fail)s;
-                    }
-                    // Because PyArray_%(func)s returns a newly created reference on t.
-                    Py_XDECREF(t);
-                }
-            """ % locals()
-        else:
-            code = """
-                if(!(%(z)s && PyArray_CompareLists(PyArray_DIMS(%(z)s), PyArray_DIMS(%(x)s), PyArray_NDIM(%(x)s))))
+                else if(axis != NPY_MAXDIMS && !(%(z)s && PyArray_CompareLists(PyArray_DIMS(%(z)s), PyArray_DIMS(%(x)s), PyArray_NDIM(%(x)s))))
```
Comment (notoraptor) [inline theano/tensor/extra_ops.py]: Here also, if z and x don't have the same number of dimensions, then some extra-memory cases may be unfortunately read by `PyArray_CompareLists()`.

**NB**: There are some functions like `theano_size_check()` and `theano_prep_output()` into new GPU backend. Should we think about implementing equivalent functions for CPU codes ?

File: theano/tensor/extra_ops.py [hunk 3]
```diff
@@ -311,57 +320,51 @@ def c_code(self, node, name, inames, onames, sub):
...
                 {
 
-                    PyObject * t = PyArray_%(func)s(
-                        %(x)s, %(axis)s,
-                        PyArray_TYPE((PyArrayObject*) py_%(x)s), %(z)s);
+                    PyObject * t = NULL;
+                    if(%(params)s->mode == MODE_ADD)
+                        t = PyArray_CumSum(
+                            %(x)s, axis,
+                            PyArray_TYPE((PyArrayObject*) py_%(x)s), %(z)s);
```
Comment (notoraptor) [inline theano/tensor/extra_ops.py]: `PyArray_TYPE((PyArrayObject*) py_%(x)s)` -> `PyArray_TYPE((x)s)` (as you do above!)

File: theano/tensor/extra_ops.py [hunk 4]
```diff
@@ -311,57 +320,51 @@ def c_code(self, node, name, inames, onames, sub):
...
+                    else if(%(params)s->mode == MODE_MUL)
+                        t = PyArray_CumProd(
+                            %(x)s, axis,
+                            PyArray_TYPE((PyArrayObject*) py_%(x)s), %(z)s);
```
Comment (notoraptor) [inline theano/tensor/extra_ops.py]: `PyArray_TYPE((PyArrayObject*) py_%(x)s)` -> `PyArray_TYPE((x)s)`

File: theano/tensor/extra_ops.py (hunk 1 above)
Comment (nouiz) [inline theano/tensor/extra_ops.py]: Above, like 327, I check cof the ndim==1. If it is this case, I set axis=NPY_MAXDIMS. So it should perform as before.

File: theano/tensor/extra_ops.py (hunk 2 above)
Comment (nouiz) [inline theano/tensor/extra_ops.py]: preallocated output should have the same ndim as what was declared in the make_node. If axis != NPY_MAXDIMS, the output ndim will be the same as the inputs ndim. So we are safe.

File: theano/tensor/extra_ops.py (hunk 3 above)
Comment (nouiz) [inline theano/tensor/extra_ops.py]: done

File: theano/tensor/extra_ops.py (hunk 4 above)
Comment (nouiz) [inline theano/tensor/extra_ops.py]: done

File: theano/tensor/extra_ops.py (hunk 1 above)
Comment (notoraptor) [inline theano/tensor/extra_ops.py]: Axis can be NPY_MAXDIMS if `self.axis` was None in Python code, and in such case I think tensor may have more than 1 dimension, isn't it ?

File: theano/tensor/extra_ops.py (hunk 2 above)
Comment (notoraptor) [inline theano/tensor/extra_ops.py]: OK!

File: theano/tensor/extra_ops.py (hunk 1 above)
Comment (notoraptor) [inline theano/tensor/extra_ops.py]: **NB**: OK, seen `make_node()`, everything is right!

**Pull request merged**